aten/src/ATen/core/SingletonSymNodeImpl.h

```diff
@@ -177,6 +234,11 @@ class TORCH_API SingletonSymNodeImpl : public SymNodeImpl {
  private:
   int64_t val_;
   int64_t coeff_;
+  at::Tensor vec_;
```

Comment:
I remember when we initially designed out singleton integers, we chose not to store the Tensor on the singleton int itself, instead opting to have some indirection managed in Python. With this PR, we are back to storing the Tensor on the singleton int. As you have discovered, this has Consequences(TM). I'd like to understand better why this must be done, and if there isn't some other approach. A not particularly well considered alternative is to store a PyInterpreter on the singleton sym int, so that we can find the right interpreter, without actually having to store the entire Tensor.

If we do go this route, we ought to get rid of the val field. But all in all, this probably should have been done in its own PR.

Reply:
Adding the Tensor onto the singleton here was useful because I need that information today during factory function. But yeah another alternative that would probably work is to continue doing all the tracking on the Python side, e.g. having a map that maps singleton id to all the singleton related metadata (offsets Tensor, min/max_seq_len).

Two reasons to store the Tensor on the singleton itself though:

1) I think the preferred solution in the long term IS to store the tensor on the singleton int itself. This would enable eventually getting rid of the val field and the extra bookkeeping on the python side (which gets more annoying to do if you want to create slices of your offsets for example). I think the reason we chose not to store the Tensor on the singleton was because  we imagined those use cases to be rare and haven't really prioritized doing it, but since it would simplify this PR, might as well bite the bullet and do it (also I already did the first step of moving singleton from c10 to aten lol).

2) Allowing a Tensor to be stored on singleton allows us to most straightforwardly support the use case of returning sizes on C++ NestedTensor (see next PR). Being able to return sizes on the C++ NestedTensor enables NT to use the same backward formulas as ordinary tensors. Today this is kind of problematic and blocks us from implementing backwards for several jagged layout NT ops (see this [doc](https://docs.google.com/document/d/1DXu-eDrfk5AYXP-p_pnh9hkyNvAq6WZNw5KJ9QdYrdg/edit#heading=h.9hmf9uv9p3wr)).